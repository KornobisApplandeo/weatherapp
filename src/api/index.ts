import {API_KEY} from '@env';

export type IWeatherResponse = {
  list: WeatherData[];
};

export type WeatherData = {
  main: Main;
  dt_txt: string;
  weather: Weather[];
};

type Main = {
  temp: number;
  feels_like: number;
  temp_min?: number;
  temp_max?: number;
  pressure?: number;
  sea_level?: number;
  grnd_level?: number;
  humidity?: number;
  temp_kf?: number;
};

type Weather = {
  id?: number;
  main?: string;
  description: string;
  icon?: string;
};

export const searchForWeather = async (city: string) => {
  try {
    const response = await fetch(
      `http://api.openweathermap.org/data/2.5/forecast?appid=${API_KEY}&q=${city}&cnt=5`,
    );
    return response;
  } catch (e) {
    console.log('error:', e);
  }
};
