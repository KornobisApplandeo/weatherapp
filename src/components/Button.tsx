import React, {FC} from 'react';
import {
  ActivityIndicator,
  StyleProp,
  StyleSheet,
  Text,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import {colors, FontFamily} from '../types/enums';

interface Props {
  title: string;
  isLoading: boolean;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
}

const Button: FC<Props> = ({title, isLoading, style, onPress}) => {
  return (
    <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
      {isLoading ? (
        <ActivityIndicator />
      ) : (
        <Text style={styles.text}>{title}</Text>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 5,
    backgroundColor: colors.orange,
    padding: 15,
    width: 200,
    alignItems: 'center',
    shadowOffset: {
      width: 20,
      height: 20,
    },
  },
  text: {
    fontFamily: FontFamily.ROBOTO_BOLD,
    color: colors.white,
    fontSize: 12,
  },
});

export default Button;
