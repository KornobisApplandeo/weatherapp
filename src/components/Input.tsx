import React, {FC} from 'react';
import {TextInput, StyleProp, TextStyle} from 'react-native';

interface Props {
  onChangeText: (text: string) => void;
  style: StyleProp<TextStyle>;
}

const BaseTextInput: FC<Props> = ({onChangeText, style}) => {
  return (
    <TextInput onChangeText={onChangeText} style={style} textAlign="center" />
  );
};

export default BaseTextInput;
