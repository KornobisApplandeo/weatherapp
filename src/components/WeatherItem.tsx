import React, {FC} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {WeatherData} from '../api';
import {colors, FontFamily} from '../types/enums';
import {format} from 'date-fns';

interface Props {
  data: WeatherData;
}

const WeatherItem: FC<Props> = ({data}) => {
  return (
    <View style={styles.container}>
      <View style={styles.dateWrapper}>
        <Text style={styles.timeText}>
          {format(new Date(data.dt_txt), 'eeeeee, dd.MM.yyyy')}
        </Text>
        <Text style={styles.timeText}>
          {format(new Date(data.dt_txt), 'p')}
        </Text>
      </View>
      <Text style={styles.temperatureText}>
        temperature: {data.main.temp} K
      </Text>
      <Text style={styles.feelsLikeTempText}>
        feels like temperature: {data.main.feels_like} K
      </Text>
      <View style={styles.descriptionWrapper}>
        <Text style={styles.descriptionText}>
          {data.weather[0].description}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    borderRadius: 10,
    padding: 20,
  },
  dateWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  temperatureText: {
    fontSize: 16,
    fontFamily: FontFamily.ROBOTO_BOLD,
    paddingVertical: 5,
  },
  feelsLikeTempText: {
    fontSize: 16,
    fontFamily: FontFamily.ROBOTO_BOLD,
  },
  descriptionText: {
    fontSize: 12,
    marginRight: 8,
    fontFamily: FontFamily.ROBOTO_BOLD,
  },
  descriptionWrapper: {
    marginTop: 15,
    borderRadius: 25,
    padding: 7,
    backgroundColor: colors.light_blue,
    alignSelf: 'flex-start',
  },
  timeText: {
    fontSize: 14,
    fontFamily: FontFamily.ROBOTO_BOLD,
    color: colors.orange,
  },
});

export default WeatherItem;
