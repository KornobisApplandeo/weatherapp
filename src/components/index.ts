import Button from './Button';
import BaseTextInput from './Input';
export {Button, BaseTextInput};
