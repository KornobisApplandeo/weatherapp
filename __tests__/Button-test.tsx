import React from 'react';
import {render} from '@testing-library/react-native';
import {Button} from '../src/components';

const onPress = jest.fn();

const setup = () => {
  return render(<Button title="test" isLoading={false} onPress={onPress} />);
};

describe('Button component', () => {
  it('renders without crashing', () => {
    const component = setup().toJSON();
    expect(component).toMatchSnapshot();
  });
});
