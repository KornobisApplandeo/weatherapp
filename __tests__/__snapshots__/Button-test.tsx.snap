// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders without crashing 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F3997B",
      "borderRadius": 5,
      "opacity": 1,
      "padding": 15,
      "shadowOffset": Object {
        "height": 20,
        "width": 20,
      },
      "width": 200,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Roboto-Bold",
        "fontSize": 12,
      }
    }
  >
    test
  </Text>
</View>
`;
